<launch>
    <!-- the number of frames accumulated to do the calibration, default:60 -->
    <arg name="ns_s1" default="depth_camera"/>
    <arg name="ns_s2" default="rgb"/>
    <arg name="max_frame" default="20"/>
    <arg name="l2c_calib" default="true"/>
    <arg name="l2l_calib" default="false"/>
    <arg name="rgb_camera_name" default="vmpr_camera"/>
    <arg name="depth_camera_name" default="vmpr_camera"/>
    <arg name="save_final_data" default="false"/>
    <arg name="rgb_depth_detection_timeout" default="45.0"/>

    <group if="$(arg l2c_calib)">
        <include file="$(find lvt2calib)/launch/pattern_collection/pattern_collection_lc.launch">
            <arg name="ns_l" value="$(arg ns_s1)"/>
            <arg name="ns_c" value="$(arg ns_s2)"/>
            <arg name="max_frame" value="$(arg max_frame)"/>
            <arg name="rgb_camera_name" value="$(arg rgb_camera_name)"/>
            <arg name="depth_camera_name"  value="$(arg depth_camera_name)"/>
            <arg name="save_final_data" value="$(arg save_final_data)"/>
            <arg name="rgb_depth_detection_timeout" value="$(arg rgb_depth_detection_timeout)"/>
        </include>
    </group>

    <group if="$(arg l2l_calib)">
        <include file="$(find lvt2calib)/launch/pattern_collection/pattern_collection_ll.launch">
            <arg name="ns_l1" value="$(arg ns_s1)"/>
            <arg name="ns_l2" value="$(arg ns_s2)"/>
            <arg name="max_frame" value="$(arg max_frame)"/>
        </include>
    </group>
</launch>